src/tools/genpolicy/genpolicy-settings.json

```diff
@@ -263,19 +263,19 @@
     "request_defaults": {
         "CreateContainerRequest": {
             "allow_env_regex": [
-                "^HOSTNAME=$(dns_label)$",
-                "^$(svc_name)_PORT_$(ip_p)_TCP=tcp://$(ipv4_a):$(ip_p)$",
-                "^$(svc_name)_PORT_$(ip_p)_TCP_PROTO=tcp$",
-                "^$(svc_name)_PORT_$(ip_p)_TCP_PORT=$(ip_p)$",
-                "^$(svc_name)_PORT_$(ip_p)_TCP_ADDR=$(ipv4_a)$",
-                "^$(svc_name)_SERVICE_HOST=$(ipv4_a)$",
-                "^$(svc_name)_SERVICE_PORT=$(ip_p)$",
-                "^$(svc_name)_SERVICE_PORT_$(dns_label)=$(ip_p)$",
-                "^$(svc_name)_PORT=tcp://$(ipv4_a):$(ip_p)$",
-                "^AZURE_CLIENT_ID=[A-Fa-f0-9-]+$",
-                "^AZURE_TENANT_ID=[A-Fa-f0-9-]+$",
-                "^AZURE_FEDERATED_TOKEN_FILE=/var/run/secrets/azure/tokens/azure-identity-token$",
-                "^AZURE_AUTHORITY_HOST=https://login\\.microsoftonline\\.com/$"
+                "^HOSTNAME=$(dns_label)?$",
+                "^$(svc_name)_PORT_$(ip_p)_TCP=(tcp://$(ipv4_a):$(ip_p))?$",
+                "^$(svc_name)_PORT_$(ip_p)_TCP_PROTO=(tcp)?$",
+                "^$(svc_name)_PORT_$(ip_p)_TCP_PORT=$(ip_p)?$",
+                "^$(svc_name)_PORT_$(ip_p)_TCP_ADDR=$(ipv4_a)?$",
+                "^$(svc_name)_SERVICE_HOST=$(ipv4_a)?$",
+                "^$(svc_name)_SERVICE_PORT=$(ip_p)?$",
+                "^$(svc_name)_SERVICE_PORT_$(dns_label)=$(ip_p)?$",
```

Comment:
You probably need to wrap the variables in parentheses to account for variable expansion, e.g. `($(ip_p))?`

Reply:
Updated to only make AZURE_TENANT_ID and AZURE_CLIENT_ID optional after talking to Dan https://github.com/microsoft/kata-containers/pull/120/files#diff-70c6543e7bd3a61dae6d1fdd1d50996e12974e137fcb3d47d36faa16db1e2f3b